src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5548,7 +5584,7 @@ bool MergeTreeData::moveParts(const CurrentlyMovingPartsTaggerPtr & moving_tagge
             throw;
         }
     }
-    return true;
+    return result;
```

Comment:
At first iteration some tests failed with exception "Cannot move parts because moves are manually disabled". This exception throws from MergeTreeData::movePartitionToVolume and MergeTreeData::movePartitionToDisk when MergeTreeData::movePartsToSpace (with "return moveParts" inside) returns false. I forget which test failed with this error but looks like it should be fail in current version too. But now all tests passed (except one timeout). Is's stange.

Reply:
I missed that failed test was for non-replicated MergeTree, it's fixed now.

Reply:
Discussed in TG problem fixed by `supportsReplication` flag.